src/box/lua/upgrade.lua

```diff
@@ -1942,10 +1958,41 @@ local function drop_instance_names(issue_handler)
     box.space._space:update({box.schema.CLUSTER_ID}, {{'=', 7, format}})
 end
 
+local function drop_trigger_from_func(issue_handler)
+    local _space = box.space[box.schema.SPACE_ID]
+    local _func = box.space[box.schema.FUNC_ID]
+    local _vfunc = box.space[box.schema.VFUNC_ID]
+    local msg_suffix = 'has trigger option. It is supported from version 3.0.0'
+    if #_func:format() == 19 then
+        return
+    end
+    if issue_handler.dry_run then
+        for _, v in _func:pairs() do
+            -- Non-empty trigger option cannot be dropped because
+            -- func does not support alter.
+            if #v > 19 and not table.equals(v[20], {}) then
+                issue_handler('Func %s %s', v.name, msg_suffix)
+            end
+        end
+        return
+    end
+    log.info('drop trigger from _func')
+    local ops = {{'#', '[7][20]', 1}}
+    _space:update({_func.id}, ops)
+    _space:update({_vfunc.id}, ops)
+    for _, v in _func:pairs() do
+        if #v > 19 and not table.equals(v[20], {}) then
```

Comment:
Having this check in two loops looks ugly. Better check the dry run flag in one loop:

```lua
for ... do
    if ... then
        issue_handler(...)
    end
    if not dry_run then
        ...
    end
end
```

Reply:
Done.